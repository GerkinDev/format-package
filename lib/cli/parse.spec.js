jest.mock('yargs', () => {
  const mock = {
    command: jest.fn(() => mock),
    options: jest.fn(() => mock),
    help: jest.fn(() => mock),
    parse: jest.fn(v => v),
  };
  return mock;
});

const yargs = require('yargs');
const parse = require('./parse');

describe('parse', () => {
  it('parses args with yargs', () => {
    expect(parse('argv')).toEqual('argv');
    expect(yargs.command.mock.calls[0]).toMatchSnapshot();
    expect(yargs.options.mock.calls[0][0]).toMatchSnapshot();
    expect(yargs.help).toHaveBeenCalledWith('h');
  });
});
