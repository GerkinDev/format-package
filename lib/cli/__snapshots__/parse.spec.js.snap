// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse parses args with yargs 1`] = `
Array [
  Array [
    "format [file]",
    "*",
  ],
  "Format closest package.json file",
]
`;

exports[`parse parses args with yargs 2`] = `
Object {
  "c": Object {
    "alias": "config",
    "demandOption": false,
    "describe": "Location of a config file",
    "type": "string",
  },
  "v": Object {
    "alias": "verbose",
    "default": false,
    "demandOption": false,
    "describe": "Flag to make the script print the formatted result",
    "type": "boolean",
  },
  "w": Object {
    "alias": "write",
    "default": false,
    "demandOption": false,
    "describe": "Flag to write the output of ordering to the package.json file",
    "type": "boolean",
  },
}
`;
